### PR TITLE
[Merged by Bors] - feat: port core Lean 3 init.data.bool.{basic,lemmas}

### PR DESCRIPTION
**Pull request by kbuzzard**
Because Lean 3 `band` and `and` have become Lean 4 `and` and `And`, and because of Lean 4 naming conventions, there are various name clashes between booly and proppy lemmas (e.g. `and_assoc` in Lean 4 is about Props). I resolve this by putting all the booly lemmas into the Bool namespace. In this PR I port all core Lean 3 bool lemmas which aren't in core Lean 4 into mathlib4, and do a lot of #align ing (including aligning core Lean 3 lemmas with their corresponding core Lean 4 names). This PR is a prerequisite for a port of mathlib3 data.bool.basic, which I have also completed and will PR next, but it seemed to me that this PR already had the possibility of generating some discussion points so I thought I'd try this first.

Some core Lean 3 lemma names are quite idiosyncratic but I've faithfully copied them over mindlessly, at least in this initial commit.

### CODE REVIEW

File: Mathlib/Init/Data/Bool/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+-/
+
+/-- Boolean XOR -/
+@[inline]
+def xor : Bool → Bool → Bool
+  | true, false => true
+  | false, true => true
+  | _, _ => false
+
+#align bxor xor
```
Comment (semorrison) [inline Mathlib/Init/Data/Bool/Basic.lean:24]: Can you remove the section from `Mathlib.Init.Align` corresponding to this file, moving its contents here instead?

Comment (semorrison) [reply Mathlib/Init/Data/Bool/Basic.lean:28]: Did it myself.

**Comment by semorrison** [comment]
I've moved all the `#align`s to come right after the statements. I think this makes life easier for readers than having a big pile of `#align`s at the bottom.

File: Mathlib/Init/Data/Bool/Basic.lean (hunk 1 above)
Comment (kbuzzard) [reply Mathlib/Init/Data/Bool/Basic.lean:28]: Thanks!

**Comment by kbuzzard** [comment]
Thanks for this Scott -- this is exactly the sort of reason I PR'ed this before mathlib's data.bool.basic. I'll put all the #aligns directly after the theorems in further PRs.

File: Mathlib/Init/Data/Bool/Lemmas.lean [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+/-!
+# Lemmas about booleans
+
+These are the lemmas about booleans which were present in core Lean 3. See also
+the file Mathlib.Data.Bool.Basic which contains lemmas about booleans from
+mathlib 3.
+
+-/
+
+-- We align Lean 3 lemmas with lemmas `Init.SimpLemmas` in Lean 4.
```
Comment (kbuzzard) [inline Mathlib/Init/Data/Bool/Lemmas.lean:19]:
```suggestion
-- We align Lean 3 lemmas with lemmas in `Init.SimpLemmas` in Lean 4.
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**